### PR TITLE
ENTESB-17767 Integration with MongoDB connector is not able to start

### PR DESCRIPTION
**Pull request by claudio4j**
https://issues.redhat.com/browse/ENTESB-17767

* Updated mongodb java driver as camel mongodb component requires it
* Updated to camel 2.23.2 fuse 7.10
* For some unknown reason there should be a wait time before connecting to mongodb embedded server

### CODE REVIEW

File: app/extension/bom/pom.xml [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
   <properties>
     <spring-boot.version>2.3.12.RELEASE</spring-boot.version>
-    <camel.version>2.23.2.fuse-790053</camel.version>
+    <camel.version>2.23.2.fuse-7_10_0-00016</camel.version>
```
Comment (zregvart) [inline app/extension/bom/pom.xml:31]: Let's do this in a separate PR, seems that we're hitting ENTESB-16583

Comment (claudio4j) [reply app/extension/bom/pom.xml:31]: Unfortunately these two should be together, if syndesis uses 2.23.2.fuse-790053, then we should use `com.mongodb.MongoClient` in mongodb connector. Also, there are tests failures, because camel component wants to use the `com.mongodb.MongoClient`.

Comment (zregvart) [reply app/extension/bom/pom.xml:31]: Chicken or the egg, well, we can't do this upstream it seams. Not unless the PR for ENTESB-16583 gets merged and all dependencies get pushed to early access.

**Comment by claudio4j** [comment]
Pushed a change to override xstream to use a version from a public repo.

**Comment by claudio4j** [comment]
Putting in draft mode to wonder how to use a fuse camel with non public artifacts.
